File: SingleTodoObjectiveC/SingleTodo/ILNewTaskViewController.m
//
//  ILNewTaskViewController.m
//  SingleTodo
//
//  Created by Abraham Kuri on 6/3/14.
//  Copyright (c) 2014 icalialabs. All rights reserved.
//

#import "ILNewTaskViewController.h"
#import "ILTodoTableViewController.h"
#import "Task.h"


@interface ILNewTaskViewController ()
@property (weak, nonatomic) IBOutlet UITextField *descriptionTextField;

@end

@implementation ILNewTaskViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)addTaskButtonPressed:(id)sender
{
    Task *task = [[Task alloc] init];
    task.description = self.descriptionTextField.text;
    task.done = NO;
    
    [self.todoVC AddTask:task];
    
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: SingleTodoSwift/SingleTodo/ILNewTaskViewController.swift
//
//  ILNewTaskViewController.swift
//  SingleTodo
//
//  Created by Abraham Kuri on 6/3/14.
//  Copyright (c) 2014 Icalia Labs. All rights reserved.
//

import UIKit

class ILNewTaskViewController: UIViewController {

    @IBOutlet var taskDescriptionTextField : UITextField
    var todoVC: ILTodoTableViewController = ILTodoTableViewController()

    @IBAction func addTaskButtonPressed(sender : AnyObject) {
        var task: ILTask = ILTask(description: self.taskDescriptionTextField.text)

        self.todoVC.addTask(task)
        
        self.navigationController.popViewControllerAnimated(true)
    }

}
